test/test_jit.py

```diff
@@ -14869,16 +14869,28 @@ def invoke_function():
         for a, b in zip(compiled_fns_1, compiled_fns_2):
             self.assertIs(a, b)
 
+        old_func = test_simple
+
+        # testing that new functions added work with caching
+        @torch.jit._overload  # noqa: F811
+        def test_simple(x1):  # noqa: F811
+            # type: (str) -> str
+            pass
+
+        @torch.jit.script
+        def my_func():
+            return old_func("hi")
+
         # currently we take the default values have to be specified in the
         # overload as well - TODO take them from implementation and apply
         # where the type is valid.
         @torch.jit._overload  # noqa: F811
         def identity(x1):  # noqa: F811
             # type: (str) -> str
             pass
-
+        #
         @torch.jit._overload  # noqa: F811
-        def identity(x1=1.0):  # noqa: F811
+        def identity(x1):  # noqa: F811
```

Comment:
bc i updated the behavior so that it errors when default arguments are put on overload declarations, which is the correct behavior. If JIT were to take them then there would be a divergence in behavior between pytorch eager & JIT, since the overloads have no runtime effect